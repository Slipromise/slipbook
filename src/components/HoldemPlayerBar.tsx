import numeral from "numeral";
import React from "react";
import Container from "react-bootstrap/esm/Container";
import Stack from "react-bootstrap/esm/Stack";
import styles from "../styles/components/HoldemPlayerBar.module.scss";

type PokerCard =
  | "SA"
  | "S2"
  | "S3"
  | "S4"
  | "S5"
  | "S6"
  | "S7"
  | "S8"
  | "S9"
  | "ST"
  | "SJ"
  | "SQ"
  | "SK"
  | "HA"
  | "H2"
  | "H3"
  | "H4"
  | "H5"
  | "H6"
  | "H7"
  | "H8"
  | "H9"
  | "HT"
  | "HJ"
  | "HQ"
  | "HK"
  | "DA"
  | "D2"
  | "D3"
  | "D4"
  | "D5"
  | "D6"
  | "D7"
  | "D8"
  | "D9"
  | "DT"
  | "DJ"
  | "DQ"
  | "DK"
  | "CA"
  | "C2"
  | "C3"
  | "C4"
  | "C5"
  | "C6"
  | "C7"
  | "C8"
  | "C9"
  | "CT"
  | "CJ"
  | "CQ"
  | "CK";

type Props = {
  avatarUri: string;
  chipAmount: number;
  name: string;
  status: string;
  cards: [PokerCard, PokerCard];
  position: "BB" | "SB" | "D" | "CO" | "UTG";
  isTurn: boolean;
};

function HoldemPokerPlayerBar({
  avatarUri,
  chipAmount,
  name,
  status = "",
  cards,
  position,
}: Props) {
  return (
    <Container className={styles.container}>
      <Stack data-left>
        <img src={avatarUri} alt="" />
        <span>{numeral(chipAmount).format("0a", Math.floor)}</span>
      </Stack>
      <Stack data-middle>
        <div data-cards>
          <span data-card={cards[0]}>{cards[0][1]}</span>
          <span data-card={cards[1]}>{cards[1][1]}</span>
        </div>
        <div data-name-container>
          <span>{name}</span>
        </div>
        <div data-status-container>
          <span>{status}</span>
        </div>
      </Stack>
      <span data-position={position}>{position}</span>
    </Container>
  );
}

export default HoldemPokerPlayerBar;
