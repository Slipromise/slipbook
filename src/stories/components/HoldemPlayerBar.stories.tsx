import type { Meta, StoryObj } from "@storybook/react";

import HoldemPlayerBar from "../../components/HoldemPlayerBar";
import { run as runHolder } from "holderjs";
import { useEffect } from "react";

const meta: Meta<typeof HoldemPlayerBar> = {
  component: HoldemPlayerBar,
  decorators: [
    (SC) => {
      useEffect(() => {
        runHolder();
      }, []);
      return <SC></SC>;
    },
  ],
};

export default meta;

type Story = StoryObj<typeof HoldemPlayerBar>;

export const Normal: Story = {
  args: {
    avatarUri: "holder.js/30x40?auto=yes&random=yes",
    chipAmount: 123944320,
    name: "PlayerName",
    status: "CHECK",
    cards: ["C2", "H7"],
    position: "D",
  },
};

export const InTurn: Story = {
  args: {
    avatarUri: "holder.js/30x40?auto=yes&random=yes",
    chipAmount: 123944320,
    name: "PlayerName",
    status: "CHECK",
    cards: ["SA", "DJ"],
    position: "D",
    isTurn: true,
  },
};

export const Cover: Story = {
  args: {
    avatarUri: "holder.js/30x40?auto=yes&random=yes",
    chipAmount: 123944320,
    name: "PlayerName",
    status: "CHECK",
    cards: [undefined, undefined],
    position: "D",
  },
};

export const Fold: Story = {
  args: {
    avatarUri: "holder.js/30x40?auto=yes&random=yes",
    chipAmount: 123944320,
    name: "PlayerName",
    status: "FOLD",
    cards: ["CK", "HT"],
    position: "D",
  },
};
