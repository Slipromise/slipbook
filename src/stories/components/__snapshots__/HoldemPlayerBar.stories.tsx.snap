// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/HoldemPlayerBar Cover smoke-test 1`] = `
<div class="_container_2rwrs_6 container _container_2rwrs_6-appear _container_2rwrs_6-appear-active">
  <div data-left="true"
       class="vstack"
  >
    <img src="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2230%22%20height%3D%2240%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2030%2040%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_18f57f21c08%20text%20%7B%20fill%3A%231E292C%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A10pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_18f57f21c08%22%3E%3Crect%20width%3D%2230%22%20height%3D%2240%22%20fill%3D%22%2339DBAC%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22-3.5390625%22%20y%3D%2232%22%3E30x40%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
         alt
         class
    >
    <span>
      123M
    </span>
  </div>
  <div data-middle="true"
       class="vstack"
  >
    <div data-cards="true">
      <div class="_container_orzyj_2"
           data-card
      >
        <span data-front-card="true">
        </span>
        <span data-back-card="true">
        </span>
      </div>
      <div class="_container_orzyj_2"
           data-card
      >
        <span data-front-card="true">
        </span>
        <span data-back-card="true">
        </span>
      </div>
    </div>
    <div data-name-container="true">
      <span>
        PlayerName
      </span>
    </div>
    <div data-status-container="true">
      <span>
        CHECK
      </span>
    </div>
  </div>
  <span data-position="D">
    D
  </span>
</div>
`;

exports[`stories/components/HoldemPlayerBar Fold smoke-test 1`] = `""`;

exports[`stories/components/HoldemPlayerBar InTurn smoke-test 1`] = `
<div data-is-turn="true"
     class="_container_2rwrs_6 container _container_2rwrs_6-appear _container_2rwrs_6-appear-active"
>
  <div data-left="true"
       class="vstack"
  >
    <img src="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2230%22%20height%3D%2240%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2030%2040%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_18f57f21c08%20text%20%7B%20fill%3A%231E292C%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A10pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_18f57f21c08%22%3E%3Crect%20width%3D%2230%22%20height%3D%2240%22%20fill%3D%22%2339DBAC%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22-3.5390625%22%20y%3D%2232%22%3E30x40%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
         alt
         class
    >
    <span>
      123M
    </span>
  </div>
  <div data-middle="true"
       class="vstack"
  >
    <div data-cards="true">
      <div class="_container_orzyj_2"
           data-card="SA"
      >
        <span data-front-card="true">
          A
        </span>
        <span data-back-card="true">
        </span>
      </div>
      <div class="_container_orzyj_2"
           data-card="DJ"
      >
        <span data-front-card="true">
          J
        </span>
        <span data-back-card="true">
        </span>
      </div>
    </div>
    <div data-name-container="true">
      <span>
        PlayerName
      </span>
    </div>
    <div data-status-container="true">
      <span>
        CHECK
      </span>
    </div>
  </div>
  <span data-position="D">
    D
  </span>
</div>
`;

exports[`stories/components/HoldemPlayerBar Normal smoke-test 1`] = `
<div class="_container_2rwrs_6 container _container_2rwrs_6-appear _container_2rwrs_6-appear-active">
  <div data-left="true"
       class="vstack"
  >
    <img src="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2230%22%20height%3D%2240%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2030%2040%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_18f57f21c08%20text%20%7B%20fill%3A%231E292C%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A10pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_18f57f21c08%22%3E%3Crect%20width%3D%2230%22%20height%3D%2240%22%20fill%3D%22%2339DBAC%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22-3.5390625%22%20y%3D%2232%22%3E30x40%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
         alt
         class
    >
    <span>
      123M
    </span>
  </div>
  <div data-middle="true"
       class="vstack"
  >
    <div data-cards="true">
      <div class="_container_orzyj_2"
           data-card="C2"
      >
        <span data-front-card="true">
          2
        </span>
        <span data-back-card="true">
        </span>
      </div>
      <div class="_container_orzyj_2"
           data-card="H7"
      >
        <span data-front-card="true">
          7
        </span>
        <span data-back-card="true">
        </span>
      </div>
    </div>
    <div data-name-container="true">
      <span>
        PlayerName
      </span>
    </div>
    <div data-status-container="true">
      <span>
        CHECK
      </span>
    </div>
  </div>
  <span data-position="D">
    D
  </span>
</div>
`;
